data/mil2525d/utilities/military-features-utilities/Bc2_and_C_CSV_generation_for_domains/Script/GenerateDomainCSV.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2016 Esri
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+# GenerateDomainCSVs.py
+# Description: Reads the All_ID_Mapping_Original CSV file and
+# selects rows with Legacy Key values of 10 characters, generates
+# CSV file for the generation of 2525C or 2525Bc2 geodatabase domains.
+# Requirements: ArcGIS Desktop
+# ----------------------------------------------------------------------------------
+
+# Import arcpy module
+import arcpy, os, sys, traceback
+import csv as csv
+
+
+# variables:
+filter_codes_table_Copy = "in_memory\\filter_codes_table_Copy"
+All_ID_Mapping_Original_Copy = "in_memory\\All_ID_Mapping_Original_Copy"
+All_ID_Mapping_Sel_10_char = "in_memory\\All_ID_Mapping_Sel_10_char"
+CharlieName = ""
+
+### FOR TESTING ###choose_standard = "C"   # permissible values are B2 or C
+### FOR TESTING ###filter_codes_table = "C:\\AutomateProcess\\Input\\Filter_Codes.csv"
+### FOR TESTING ###All_ID_Mapping_Original_csv = "C:\\AutomateProcess\\Input\\All_ID_Mapping_Original.csv"
+### FOR TESTING ###out_domain_location = "C:\\AutomateProcess\\Output\\"
+
+def load_table_in_memory():
+    # Copy Rows
+    arcpy.env.workspace = "in_memory"
+    arcpy.CopyRows_management(All_ID_Mapping_Original_csv, All_ID_Mapping_Original_Copy, "")
+
+    # Table Select
+    if choose_standard == "B2":
+        print "Excluding C tagged items."
+        arcpy.TableSelect_analysis(All_ID_Mapping_Original_Copy, All_ID_Mapping_Sel_10_char, "(CHAR_LENGTH ( \"LegacyKey\" ) = 10) AND (\"Standard\" <> 'C') OR \"Standard\" IS Null")
+    else:
+        print "Excluding B2 tagged items."
+        arcpy.TableSelect_analysis(All_ID_Mapping_Original_Copy, All_ID_Mapping_Sel_10_char, "(CHAR_LENGTH ( \"LegacyKey\" ) = 10) AND (\"Standard\" <> 'B2') OR \"Standard\" IS Null")
+
+    # Load filter_codes_table to in_memory workspace
+    arcpy.CopyRows_management(filter_codes_table, filter_codes_table_Copy, "")
+
+def loop_filter_codes():
+    # Iterate filter_codes_table_Copy to get file names, CharlieName, Filter1, Filter2, FilterGeom values
+    arcpy.AddMessage("Iterating filter code table.")
+    fields = ['CharlieName', 'Filter1', 'Filter2', 'FilterGeom']
+    with arcpy.da.SearchCursor(filter_codes_table_Copy, fields) as cursor:
+        for row in cursor:
+            if row[3] in ["Point", "Line","Area"]:
```

Comment:
This logic below seems a little more complex than necessary, seems like you could have built the query up (and it would have been more readable/understandable):
```
query = {custom query with row[1]
if row[2] not None:
   query = query + " OR {custom query with row[2]"
if row[3] not None:
   query = query + " AND {custom query with row[3]"
```